tests/Doctrine/Tests/ORM/Functional/Ticket/GH10063Test.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\GeneratedValue;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @requires PHP 8.1
+ */
+final class GH10063Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->createSchemaForModels(GH10063Entity::class);
+    }
+
+    public function testArrayOfEnums(): void
+    {
+        $entity = (new GH10063Entity())->setColors([GH10063Enum::Red, GH10063Enum::Green]);
+
+        $this->_em->persist($entity);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $entity = $this->_em->find(GH10063Entity::class, $entity->id);
+        assert($entity instanceof GH10063Entity);
+        self::assertEquals([GH10063Enum::Red, GH10063Enum::Green], $entity->getColors());
+    }
+}
+
+/**
+ * @Entity
+ * @requires PHP 8.1
+ */
+class GH10063Entity
+{
+    /**
+     * @Column
+     * @Id
+     * @GeneratedValue
+     */
+    public int $id;
```

Comment:
Code using recent syntax should be moved to a separate file, otherwise PHP 7.1 will crash when parsing the file prior to realizing it should not execute the test.

Reply:
Well, the entire test is about enums (i.e. recent syntax). So how can I tell PHP 7 to ignore it?

Reply:
1. Make sure `GH10063Test` can be _parsed_ by PHP 7
2. Move `GH10063Entity` and `GH10063Enum` to a separate file.